File: 36-valid-sudoku/valid-sudoku.java
import java.util.Arrays;
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<String> set = new HashSet<>();
        for(int i = 0; i < board.length; i++){
            for(int j = 0; j < board[0].length; j++){
                if(board[i][j] == '.'){
                    continue;
                }
                if(set.contains(board[i][j] + " in row" + i) || set.contains(board[i][j] + " in col" + j) || set.contains(board[i][j] + "in box" + i/3 + j/3)){
                    return false;
                }
                set.add(board[i][j] + " in row" + i);
                set.add(board[i][j] + " in col" + j);
                set.add(board[i][j] + "in box" + i/3 + j/3);
            }
        }
        return true;  
    }
}

File: 36-valid-sudoku/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        cols = set()
        rows = set()
        boxes = set()
        i = 0
        j = 0
        while i < len(board):
            j = 0
            while j < len(board[0]):
                curr = board[i][j]
                currrow = (curr, i)
                currcol = (curr, j)
                currbox = (curr, i//3, j//3)
                if curr == '.':
                    j = j + 1
                    continue
                if (currcol in cols) or (currrow in rows) or (currbox in boxes):
                    return False
                cols.add((curr, j))
                rows.add((curr, i))
                boxes.add((curr, i//3, j//3))
                j = j + 1
            i = i + 1
        return True
            

        
        